### PR TITLE
Adding support for cfndsl external parameters

### PR DESCRIPTION
**Pull request by elmobp**
To resolve this issue https://github.com/envato/stack_master/issues/143 Given I needed to access both stack_definition and config within the template compiler I have modified a few files and updated to 0.9.1 though I can revert that if required! This is a awesome piece of software I must say!

### CODE REVIEW

**Comment by elmobp** [comment]
Apologies, fixing the tests now

**Comment by gergnz** [comment]
I'm keen to see this PR merged team.

File: lib/stack_master/template_compilers/cfndsl.rb [hunk 1]
```diff
@@ -4,10 +4,12 @@ def self.require_dependencies
       require 'cfndsl'
     end
 
-    def self.compile(template_file_path)
-      ::CfnDsl.eval_file_with_extras(template_file_path).to_json
+    def self.compile(config, template_file_path, stack_definition)
+      params = []
+      params.push([:yaml, File.join(config.base_dir, 'external_parameters', stack_definition.cfndsl_external_parameters)]) if stack_definition.cfndsl_external_parameters.length > 0
```
Comment (patrobinson) [inline lib/stack_master/template_compilers/cfndsl.rb]: Can we break this line up?

Comment (elmobp) [inline lib/stack_master/template_compilers/cfndsl.rb]: Sure, ill do that today

**Comment by elmobp** [comment]
@patrobinson I not sure how what you mean by the first comment? I can move the params to stack based parms if thats what you mean

**Comment by patrobinson** [comment]
@elmobp Instead of passing the entire `stack_definition` object to TemplateCompiler, we can just send the external parameters `stack_definition.cfndsl_external_parameters`

**Comment by elmobp** [comment]
@patrobinson Sure! I am keen to see this move

**Comment by elmobp** [comment]
@patrobinson the other reason for passing it all in was if we wanted to support an extension of another compiler the whole object is already there :) but let me know thoughts

**Comment by elmobp** [comment]
@patrobinson fine with me thought we should keep it right up until compile is called? as for not using CFN param's its for this reason :) https://github.com/stevenjack/cfndsl/issues/170

**Comment by gergnz** [comment]
@patrobinson I assume by Cloudformation parameters, you mean the ones you put into Cloudformation API calls. If so the reason we use yml params at compile time of cfndsl is that we use things like iteration, and all sorts of other complex logic in ruby to build the data in the final cloudformation. It would be too late to provide those as cloudformation parameters.

File: lib/stack_master/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module StackMaster
-  VERSION = "0.9.0"
+  VERSION = "0.12.1"
```
Comment (stevehodgkiss) [inline lib/stack_master/version.rb]: This should be `0.13.0` if we're aiming to follow semantic versioning.

Comment (elmobp) [inline lib/stack_master/version.rb]: Done!

**Comment by johnf** [comment]
I would also love to see this merge @patrobinson are there further changes you would still like resolved?

Disclaimer: I also work at Bulletproof, although I'm actually chasing this up for a personal project :)

**Comment by patrobinson** [comment]
@johnf This needs a merge because it conflicts with `master` and it's a dirty branch, you can see a lot of commits not related to this PR here. Are you able to clean it up so I can give it another review?

**Comment by patrobinson** [comment]
I think this was superseeded by #120?

**Pull request closed without merging**